test/tools/ossfuzz/protomutators/YulProtoMutator.cpp

```diff
@@ -0,0 +1,2081 @@
+	else if constexpr (std::is_same_v<std::decay_t<T>, BinaryOp_BOp>)
+		return BinaryOp_BOp_BOp_MIN;
+	else if constexpr (std::is_same_v<std::decay_t<T>, UnaryOp_UOp>)
+		return UnaryOp_UOp_UOp_MIN;
+	else if constexpr (std::is_same_v<std::decay_t<T>, LowLevelCall_Type>)
+		return LowLevelCall_Type_Type_MIN;
+	else if constexpr (std::is_same_v<std::decay_t<T>, Create_Type>)
+		return Create_Type_Type_MIN;
+	else if constexpr (std::is_same_v<std::decay_t<T>, UnaryOpData_UOpData>)
+		return UnaryOpData_UOpData_UOpData_MIN;
+	else
+		static_assert(AlwaysFalse<T>::value, "Yul proto mutator: non-exhaustive visitor.");
+}
+
+Expression* YPM::loadExpression(YulRandomNumGenerator& _rand)
+{
+	auto unop = new UnaryOp();
+	unop->set_allocated_operand(refExpression(_rand));
+	switch (_rand() % 3)
+	{
+	case 0:
+		unop->set_op(UnaryOp::MLOAD);
+		break;
+	case 1:
+		unop->set_op(UnaryOp::SLOAD);
+		break;
+	case 2:
+		unop->set_op(UnaryOp::CALLDATALOAD);
+		break;
+	}
+	auto expr = new Expression();
+	expr->set_allocated_unop(unop);
+	return expr;
+}
+
+Expression* YPM::loadFromZero(YulRandomNumGenerator& _rand)
+{
+	auto unop = new UnaryOp();
+	unop->set_allocated_operand(litExpression(0));
+	switch (_rand() % 3)
+	{
+	case 0:
+		unop->set_op(UnaryOp::MLOAD);
+		break;
+	case 1:
+		unop->set_op(UnaryOp::SLOAD);
+		break;
+	case 2:
+		unop->set_op(UnaryOp::CALLDATALOAD);
+		break;
+	}
+	auto expr = new Expression();
+	expr->set_allocated_unop(unop);
+	return expr;
+}
+
+void YPM::clearExpr(Expression* _expr)
+{
+	switch (_expr->expr_oneof_case())
+	{
+	case Expression::kVarref:
+		delete _expr->release_varref();
+		_expr->clear_varref();
+		break;
+	case Expression::kCons:
+		delete _expr->release_cons();
+		_expr->clear_cons();
+		break;
+	case Expression::kBinop:
+		delete _expr->release_binop();
+		_expr->clear_binop();
+		break;
+	case Expression::kUnop:
+		delete _expr->release_unop();
+		_expr->clear_unop();
+		break;
+	case Expression::kTop:
+		delete _expr->release_top();
+		_expr->clear_top();
+		break;
+	case Expression::kNop:
+		delete _expr->release_nop();
+		_expr->clear_nop();
+		break;
+	case Expression::kFuncExpr:
+		delete _expr->release_func_expr();
+		_expr->clear_func_expr();
+		break;
+	case Expression::kLowcall:
+		delete _expr->release_lowcall();
+		_expr->clear_lowcall();
+		break;
+	case Expression::kCreate:
+		delete _expr->release_create();
+		_expr->clear_create();
+		break;
+	case Expression::kUnopdata:
+		delete _expr->release_unopdata();
+		_expr->clear_unopdata();
+		break;
+	case Expression::EXPR_ONEOF_NOT_SET:
+		break;
+	}
+}
+
+Expression* YPM::binopExpression(YulRandomNumGenerator& _rand)
+{
+	auto binop = new BinaryOp();
+	binop->set_allocated_left(refExpression(_rand));
+	binop->set_allocated_right(refExpression(_rand));
+	binop->set_op(
+		YPM::EnumTypeConverter<BinaryOp_BOp>{}.enumFromSeed(_rand())
+	);
+	auto expr = new Expression();
+	expr->set_allocated_binop(binop);
+	return expr;
+}
+
+void YPM::unsetExprMutator(
```

Comment:
What does it mean to unset the mutator in the expressions?

Reply:
Actually this one needs to be read as `mutator for unset expressions` :-)

Unset expressions are protobuf messages of type `Expression` that have not been initialized.